File: src/examples/bitbang.py
#!/usr/bin/env python

from mpsse import *
from time import sleep

# Open mpsse in bitbang mode
io = MPSSE(BITBANG);

# Set pin 0 high/low and read back pin 0's state 10 times
for i in range(0, 10):

	io.set_pin_high(0)
	print "Pin 0 is:", io.pin_state(0)
	sleep(1)

	io.set_pin_low(0)
	print "Pin 0 is:", io.pin_state(0)
	sleep(1)

io.Close()

File: src/examples/bitbang.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#include "mpsse.h"

int main(void)
{
	struct mpsse_context *io = NULL;
	int i = 0, retval = EXIT_FAILURE;

	io = mpsse_init(BITBANG, 0, 0);

	if(io && io->open)
	{
		for(i=0; i<10; i++)
		{
			set_pin_high(io, 0);
			printf("Pin 0 is: %d\n", pin_state(io, 0, -1));
			sleep(1);
			
			set_pin_low(io, 0);
			printf("Pin 0 is: %d\n", pin_state(io, 0, -1));
			sleep(1);
		}

		retval = EXIT_SUCCESS;
	}
	else
	{
		printf("Failed to open MPSSE: %s\n", error_string(io));
	}

	release(io);

	return retval;
}
